### PR TITLE
Add section on how to modify an arbitrary commit

### PR DESCRIPTION
**Pull request by m007**
Explain how to solve situations where you want to modify a commit which
is not your last by using interactive rebase.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -344,6 +345,38 @@ If you want to completely remove an entire file (and not keep it locally), then
 
 If you have made other commits in the meantime (i.e. the sensitive data is in a commit before the previous commit), you will have to rebase. 
 
+<a href="i-need-to-change-the-content-of-a-commit-which-is-not-my-last"></a>
+### I need to change the content of a commit which is not my last
+
+Consider you created some (e.g. three) commits and later realise you missed doing something that belongs contextually into the first of those commits. This bothers you because if you'd create a new commit containing those changes, you'd have a clean code base, but your commits weren't atomic (and this matters much to you, because you're a good developer). In such a situation you may want to change the commit where these changes belong to contain them and have the rest of your following commits unaltered (because they are atomic and that is because - you might guess it- you are a very good developer). In such a case, `git rebase` might safe you.
```
Comment (RichardLitt) [inline README.md]: I would leave out the parentheticals about being a good developer. Telling people that they should care about a manner of coding or they're not good developers is not an effective way to teach. Instead, highlight why atomic commits matter in a parenthesis. Can you rewrite this to be clearer, there? 

Also, some commas would be great. This paragraph was a bit hard to parse.

File: README.md [hunk 2]
```diff
@@ -344,6 +345,38 @@ If you want to completely remove an entire file (and not keep it locally), then
...
+```sh
+(your-branch)$ git rebase -i HEAD~4
+```
+gets you into interactive rebase mode, which allows you to edit any of your last three commits. A text editor pops up, showing you something like
+```sh
+pick 9e1d264 The third last commit
+pick 4b6e19a The second to last commit
+pick f4037ec The last commit
+```
+which you elegantly change into
```
Comment (RichardLitt) [inline README.md]: Remove elegantly.

File: README.md [hunk 3]
```diff
@@ -344,6 +345,38 @@ If you want to completely remove an entire file (and not keep it locally), then
...
+```sh
+edit 9e1d264 The third last commit
+pick 4b6e19a The second to last commit
+pick f4037ec The last commit
+```
+This tells rebase that you want to edit your third last commit and keep the other two unaltered. Then you'll save (and close) the editor. `git` will then start to rebase. It stops on the commit you want to alter. This is your chance to alter that commit. Now you have to fix the files you missed to alter when you initially commited that commit. You do so by editing and staging them. Afterwars you'll run
```
Comment (RichardLitt) [inline README.md]: > Now you have to fix the files you missed to alter when you initially commited that commit. 

Unclear.

Also, "afterwars" means something different. :D

File: README.md [hunk 4]
```diff
@@ -344,6 +345,38 @@ If you want to completely remove an entire file (and not keep it locally), then
...
+```sh
+(your-branch)$ git commit --amend --no-edit
+```
+which tells `git` to recreate the commit, but to no edit the commit message. Having done that, the hard part is solved
```
Comment (RichardLitt) [inline README.md]: but to no edit?

File: README.md (hunk 1 above)
Comment (m007) [reply README.md]: How about

> Consider you created some (e.g. three) commits and later realise you missed doing something that belongs contextually into the first of those commits. This bothers you, because if you'd create a new commit containing those changes, you'd have a clean code base, but your commits weren't atomic (i.e. changes that belonged to each other weren't in the same commit). In such a situation you may want to change the commit where these changes belong to, include them and have the following commits unaltered. In such a case, `git rebase` might safe you.

File: README.md (hunk 3 above)
Comment (m007) [reply README.md]: How about 

> This tells rebase that you want to edit your third last commit and keep the other two unaltered. Then you'll save (and close) the editor. `git` will then start to rebase. It stops on the commit you want to alter giving you the chance to edit that commit. Now you can apply the changes you missed applying when you initially commited that commit. You do so by editing and staging them. Afterwards you'll run

File: README.md (hunk 4 above)
Comment (m007) [reply README.md]: `--no-edit` tells git to amend the commit without editing the commit message. This made sense in the siutation where I needed to change my third last commit (the commit message already explained the changes I forgot to appla in the first place). But it usually may be more probable that you need to change the message if you had to amend the commit. Should I remove `--no-edit`?

File: README.md (hunk 1 above)
Comment (RichardLitt) [reply README.md]: Change "might safe you" to "might save you", and it sounds better. :)

File: README.md (hunk 3 above)
Comment (RichardLitt) [reply README.md]: Works for me.

File: README.md (hunk 4 above)
Comment (RichardLitt) [reply README.md]: `--no-edit` is fine. You have a grammatical issue in the text: "but to no edit the commit message."

**Comment by m007** [comment]
@RichardLitt please have a look again.

**Comment by RichardLitt** [comment]
Thank you. Merged!
